lib/Doctrine/ORM/Mapping/ManyToManyAssociationMapping.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+class ManyToManyAssociationMapping extends ToManyAssociationMapping
+{
+    public array|null $relationToSourceKeyColumns = null;
+    public array|null $relationToTargetKeyColumns = null;
+
+    /** @param mixed[] $mappingArray */
+    public static function fromMappingArray(array $mappingArray): static
+    {
+        $mapping = parent::fromMappingArray($mappingArray);
+
+        $mapping['orphanRemoval'] = isset($mapping['orphanRemoval']) && $mapping['orphanRemoval'];
```

Comment:
This line confuses me. Why are you mapping this property to itself?

Reply:
I'm just moving code around here:

https://github.com/doctrine/orm/blob/13d86957808f3fab3dfb9c681c59984a2bb4e4dd/lib/Doctrine/ORM/Mapping/ClassMetadata.php#L1685

This ensures the property ends up being a boolean, treating `null` as `false`. I will remove this line and default to `false` instead.